src/components/PopupWithForm.js

```diff
@@ -1,68 +1,82 @@
 import Popup from "./Popup.js";
+import { createCard } from "../utils/createCard.js";
+import { handleSubmit } from "../utils/handleSubmit.js";
 
 export default class PopupWithForm extends Popup {
-  constructor(popupSelector, handleFormSubmit, handleFormClose = () => {}) {
+  constructor(popupSelector, handleSubmit, formValidator) {
     super({ popupSelector });
-    this._popup = document.querySelector(popupSelector);
-    this._handleFormSubmit = handleFormSubmit;
-    this._form = this._popup.querySelector("form");
-    this._inputs = Array.from(this._form.querySelectorAll("input"));
-    this._formData = {};
-    this._handleFormClose = handleFormClose;
+    this._handleSubmit = handleSubmit;
+    this._form = this.popupElement.querySelector("form");
+    this._submitButton = this._form.querySelector("button[type='submit']");
+    this._submitButtonText = this._submitButton.textContent;
+    this._formValidator = formValidator;
   }
 
-  _focusFirstInputField() {
-    const firstInput = this._inputs[0];
+  setEventListeners() {
+    super.setEventListeners();
+    this._form.addEventListener("submit", (evt) => {
+      evt.preventDefault();
+      this.renderLoading(true);
+
+      const submitPromise = this._handleSubmit(this._getInputValues());
+
+      if (!(submitPromise instanceof Promise)) {
+        console.error("The handleSubmit method did not return a promise.");
+        this.renderLoading(false);
+        return;
+      }
+
+      submitPromise
+        .then(() => this.close())
+        .catch((err) => console.error(`Error during form submission: ${err}`))
+        .finally(() => this.renderLoading(false));
+    });
+  }
+
+  resetForm() {
+    const form = this.popupElement.querySelector("form");
+    if (form) form.reset();
+  }
+
+  open(formData = {}) {
+    super.open();
+    this._populateForm(formData);
+    this._focusFirstInput();
+    this._formValidator.resetValidation();
+  }
+
+  _focusFirstInput() {
+    const firstInput = this._form.querySelector("input");
```

Comment:
`const firstInput = this._inputs[0];` was here last time. You shouldn't have changed it